python/ray/data/_internal/datasource/tfrecords_datasource.py

```diff
@@ -122,3 +142,293 @@ def _resolve_full_path(self, relative_path):
             return f"{protocol}://{relative_path}"
 
         return relative_path
+
+
+def _convert_example_to_dict(
```

Comment:
This isn't adding any new code.

Previously, `TFRecordsDatasource` was in `ray._internal.datasource` but the utility functions were in `ray.datasource`. I'm just moving the utility functions out of `ray.datasource`.